File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        
        int left = 0;
        int right = numbers.length - 1;

        while(left < right) {
            int leftVal = numbers[left];
            int rightVal = numbers[right];

            int total = leftVal + rightVal;

            if(total < target) {
                left++;
            } else if(total > target) {
                right--;
            } else {
                return new int[] { left + 1, right + 1 };
            }

        }

        return new int[0];
    }
}


// Notes

// Utilize 2 pointers
// Left & Right
// if total < target, inc left pointer. This will make the total bigger
// if total > target, dec right pointer. This will make total smaller

// This only works because its sorted*

// NeetCode wants element
// LeetCode wants index & is []1 based

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        
        #binary serach | left/right

        left = 0
        right = len(numbers) - 1

        while left < right:
            total = numbers[left] + numbers[right]

            if total == target:
                return [left + 1, right + 1] # 1 index based

            if total > target:
                right -= 1
            else:
                left += 1
